### ISSUE TITLE
Rule Change Proposal: `component-name-in-template-casing`

### ISSUE THREAD

**Issue by gburning** [vuejs/eslint-plugin-vue#854]
**Please describe what the rule should do:**
At the moment this rule applies the casing rule globally for the entire component. But the usual way of writing components in a *.vue file is to use kebab-case in the template but PascalCase in the script part.

In short, I'd like for this rule to either be split into two or to take two parameters instead of one (one for the template and the other for the script).

**What category should the rule belong to?**
- [X] Enforces code style
- [ ] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
If we set the rule so that the template should use kebab-case and the script part PascalCase this would generate an error:
```vue
<template>
    <MyCustomComponent></MyCustomComponent>
</template>

<script>
    export default {
        components: {
            MyCustomComponent
        }
    }
</script>
```

This would be correct:
```vue
<template>
    <my-custom-component></my-custom-component>
</template>

<script>
    export default {
        components: {
            MyCustomComponent
        }
    }
</script>
```
